ios/Classes/SwiftBccmPlayerPlugin.swift

```diff
@@ -2,18 +2,31 @@ import AVKit
 import Flutter
 import GoogleCast
 import UIKit
+import Combine
 
 public class SwiftBccmPlayerPlugin: NSObject, FlutterPlugin {
+    static var cancellable: AnyCancellable? = nil
+    
     public static func register(with registrar: FlutterPluginRegistrar) {
         setupCast()
         let messenger = registrar.messenger()
         let channel = FlutterMethodChannel(name: "bccm_player", binaryMessenger: messenger)
         let instance = SwiftBccmPlayerPlugin()
         registrar.addMethodCallDelegate(instance, channel: channel)
+        registrar.addApplicationDelegate(instance)
+
         let playbackListener = PlaybackListenerPigeon(binaryMessenger: messenger)
 
         let chromecastPigeon = ChromecastPigeon(binaryMessenger: messenger)
         let playbackApi = PlaybackApiImpl(chromecastPigeon: chromecastPigeon, playbackListener: playbackListener)
+        
+        let downloaderListener = DownloaderListenerPigeon(binaryMessenger: messenger)
+        let downloader = Downloader()
+        cancellable = downloader.statusChanged.sink { event in
```

Comment:
Maybe make a DownloaderListener() which does this stuff, to keep the logic out of this file.